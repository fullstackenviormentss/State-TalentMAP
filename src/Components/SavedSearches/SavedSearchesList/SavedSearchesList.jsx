import React from 'react';
import PropTypes from 'prop-types';
import SavedSearchesListResultsCard from '../SavedSearchesListResultsCard';
import { SAVED_SEARCH_PARENT_OBJECT, MAPPED_PARAM_ARRAY } from '../../../Constants/PropTypes';
import NoSavedSearches from '../../EmptyListAlert/NoSavedSearches';

const SavedSearchesList = ({ savedSearches, goToSavedSearch, deleteSearch, cloneSavedSearch,
mappedParams }) => {
  const savedSearchArray = [];
  savedSearches.results.slice().forEach(savedSearch => (
    savedSearchArray.push(
      <SavedSearchesListResultsCard
        savedSearch={savedSearch}
        goToSavedSearch={goToSavedSearch}
        deleteSearch={deleteSearch}
        key={savedSearch.id}
        cloneSavedSearch={cloneSavedSearch}
        mappedParams={mappedParams}
        /* pass a parentClassName that we can use from the BorderedList component */
        parentClassName="parent-list-container list-transparent"
      />,
    )
  ));
  let emptyListClass = 'saved-searches-list-empty';
  if (savedSearchArray.length) { emptyListClass = ''; }
  return (
    <div className="usa-grid-full">
      {
        <div className={`usa-grid-full saved-searches-list ${emptyListClass}`}>
          {savedSearchArray.map(s => s)}
        </div>
      }
      {
        !savedSearchArray.length && <NoSavedSearches />
      }
    </div>
  );
};

SavedSearchesList.propTypes = {
  savedSearches: SAVED_SEARCH_PARENT_OBJECT.isRequired,
  goToSavedSearch: PropTypes.func.isRequired,
  deleteSearch: PropTypes.func.isRequired,
  cloneSavedSearch: PropTypes.func.isRequired,
  mappedParams: MAPPED_PARAM_ARRAY.isRequired,
};

SavedSearchesList.defaultProps = {
  mappedParams: [],
};

export default SavedSearchesList;
