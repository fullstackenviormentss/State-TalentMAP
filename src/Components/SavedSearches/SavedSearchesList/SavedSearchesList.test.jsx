import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import SavedSearchesList from './SavedSearchesList';
import searchObjectParent from '../../../__mocks__/searchObjectParent';

describe('SavedSearchesListComponent', () => {
  const props = {
    savedSearches: searchObjectParent,
    goToSavedSearch: () => {},
    deleteSavedSearch: () => {},
    deleteSearch: () => {},
    cloneSavedSearch: () => {},
    mappedParams: [],
  };
  it('is defined', () => {
    const wrapper = shallow(
      <SavedSearchesList
        {...props}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(
      <SavedSearchesList
        {...props}
      />,
    );
    expect(wrapper.instance().props.savedSearches.results[0].id)
      .toBe(searchObjectParent.results[0].id);
  });

  it('displays an alert if there are no results', () => {
    const wrapper = shallow(
      <SavedSearchesList
        {...props}
        savedSearches={{ results: [] }}
      />,
    );
    expect(wrapper.find('NoSavedSearches').exists()).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <SavedSearchesList
        {...props}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
