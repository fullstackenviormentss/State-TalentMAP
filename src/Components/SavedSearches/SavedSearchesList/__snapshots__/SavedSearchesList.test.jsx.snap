// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedSearchesListComponent matches snapshot 1`] = `
<div
  className="usa-grid-full"
>
  <div
    className="usa-grid-full saved-searches-list "
  >
    <SavedSearchListResultsCard
      cloneSavedSearch={[Function]}
      deleteSearch={[Function]}
      goToSavedSearch={[Function]}
      mappedParams={Array []}
      parentClassName="parent-list-container list-transparent"
      savedSearch={
        Object {
          "count": 0,
          "date_created": "2018-03-09T17:47:07.133624Z",
          "date_updated": "2018-03-09T17:47:07.133651Z",
          "endpoint": "/api/v1/position/",
          "filters": Object {
            "grade__code__in": "02",
            "is_domestic": "true",
            "post__cost_of_living_adjustment__gt": "0",
            "post__in": "151",
            "post__tour_of_duty__code__in": "O,Q",
            "q": "german",
            "skill__code__in": "6080,2080",
          },
          "id": 7,
          "name": "Test 33",
          "owner": "Jenny Townpost",
        }
      }
    />
  </div>
</div>
`;
